protoc_plugin/lib/src/generated/descriptor.pb.dart

```diff
@@ -765,12 +767,12 @@ class FieldDescriptorProto extends $pb.GeneratedMessage {
     ..e<FieldDescriptorProto_Label>(
         4, _omitFieldNames ? '' : 'label', $pb.PbFieldType.OE,
         defaultOrMaker: FieldDescriptorProto_Label.LABEL_OPTIONAL,
-        valueOf: FieldDescriptorProto_Label.valueOf,
+        enumValueMap: FieldDescriptorProto_Label.byValue,
         enumValues: FieldDescriptorProto_Label.values)
```

Comment:
So it seems we are already passing `<XXX>.values` into the `BuilderInfo`.

With this change we also pass in `<XXX>.byValue` into the `BuilderInfo`.

But `<XXX>.byValue` (the new thing we pass into `BuilderInfo` is defined as
```

  static final $core.Map<$core.int, XXX> byValue = $pb.ProtobufEnum.initByValue(values);
```

So actually  the `BuilderInfo` can calculate `XXX.byValue` itself from the `XXX.values` it gets passed.
So it seems we are passing redundant information here, no?

Reply:
This is because all `BuilderInfo`s need to share the `byValue` map or list for the same enum type.

Reply:
Ah, I see. So we want to avoid making N `byValue` maps/lists for the same enum.

I think the smell here is this:
* Each `Foo extends GeneratedMessage` has metadata about it in the `<Foo>.info_` getter.
* Each `FooEnum extends ProtobufEnum` does not have metadata about it in a similar `<FooEnum>.info_` getter

Instead it has these various different static variables as well as `valueOf` closure, ... that we pass around individually - instead of as one e.g. `ProtobufEnumDescriptor` object.